packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/lib_navigation/use_cases/check_permission_route_use_case.dart

```diff
@@ -0,0 +1,24 @@
+{{> licence.dart }}
+
+import '../../base/common_services/permissions_service.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../base/routers/router.dart';
+
+class CheckPermissionRouteUseCase {
+  CheckPermissionRouteUseCase(this._permissionsService);
+
+  final PermissionsService _permissionsService;
+
+  Future<void> execute({
+    required bool isLoggedIn,
+    required String routePath,
+  }) async {
+    if (!isLoggedIn && (routePath != '/' && routePath != '/login')) {
+      throw NetworkErrorModel();
```

Comment:
Is the NetworkErrorModel the most appropriate error model in this case ?

Reply:
Done